image-loader/app/ImageLoaderComponents.scala

```diff
@@ -18,15 +18,20 @@ class ImageLoaderComponents(context: Context) extends GridComponents(context, ne
 
   val store = new ImageLoaderStore(config)
   val imageOperations = new ImageOperations(context.environment.rootPath.getAbsolutePath)
-
   val notifications = new Notifications(config)
   val downloader = new Downloader()
   val uploader = new Uploader(store, config, imageOperations, notifications)
-
   val projector = Projector(config, imageOperations)
-
+  val quarantineUploader: Option[QuarantineUploader] = (config.uploadToQuarantineEnabled, config.quarantineBucket) match {
+    case (true, Some(bucketName)) =>{
+      val quarantineStore = new QuarantineStore(config)
+      Some(new QuarantineUploader(quarantineStore, config))
+    }
+    case (true, None) => throw new IllegalArgumentException("Quarantining is enabled but no bucket is configured")
```

Comment:
A nice touch would be to include the configuration names in this message. i.e. `Quarantining is enabled (xxx.yyy=true) but no bucket is configured (zzz.aaa isn't configured)`.

This would save users having to look at the source code.